src/Umbraco.Web.UI.Client/src/views/common/overlays/linkpicker/linkpicker.controller.js

```diff
@@ -29,22 +29,27 @@ angular.module("umbraco").controller("Umbraco.Overlays.LinkPickerController",
 
 	        //if we have a node ID, we fetch the current node to build the form data
 	        if ($scope.model.target.id || $scope.model.target.udi) {
-
-                //will be either a udi or an int
-                var id = $scope.model.target.udi ? $scope.model.target.udi : $scope.model.target.id;
-
-                if (!$scope.model.target.path) {
+                
+                if ($scope.model.target.mode == 'content') {
```

Comment:
Maybe it's better to still do it for media items...but specify the type when calling entityResource.getPath ? Don't know if that's possible by the way.

Similar to what I have done in this PR : https://github.com/umbraco/Umbraco-CMS/pull/2543/files